### PR TITLE
add syrkx to hipblas-test and hipblas-bench

### PR DESCRIPTION
**Pull request by amcamd**
- add syrkx, syrkx_batched and syrkx_strided_batched to hipblas-bench
- make test code similar to trmm test code
- add syrkx reference implementation in place of setting A == B and using syrk
- remove large size tests for syrkx and trmm to reduce testing time

### CODE REVIEW

File: clients/include/testing_syrkx_strided_batched.hpp [hunk 1]
```diff
@@ -53,101 +48,171 @@ hipblasStatus_t testing_syrkx_strided_batched(const Arguments& argus)
         return HIPBLAS_STATUS_SUCCESS;
     }
 
+    int           K1       = transA == HIPBLAS_OP_N ? K : N;
+    hipblasStride stride_A = lda * K1 * stride_scale;
+    hipblasStride stride_B = ldb * K1 * stride_scale;
+    hipblasStride stride_C = ldc * N * stride_scale;
```
Comment (TorreZuk) [inline clients/include/testing_syrkx_strided_batched.hpp:54]: I think this is pre-existing overflow potential, either size_t(lda) or move the stride_scale double to the front to do all math as double.

File: clients/include/testing_syrkx_batched.hpp [hunk 2]
```diff
@@ -77,87 +70,148 @@ hipblasStatus_t testing_syrkx_batched(const Arguments& argus)
...
     for(int b = 0; b < batch_count; b++)
     {
-        hA[b]  = host_vector<T>(A_size);
-        hB[b]  = host_vector<T>(B_size);
-        hC[b]  = host_vector<T>(C_size);
-        hC2[b] = host_vector<T>(C_size);
+        hA_array[b]      = host_vector<T>(A_size);
+        hB_array[b]      = host_vector<T>(B_size);
+        hC_array[b]      = host_vector<T>(C_size);
+        hC_array_copy[b] = host_vector<T>(C_size);
```
Comment (TorreZuk) [inline clients/include/testing_syrkx_batched.hpp:89]: Ask @daineAMD as maybe the host_batched_vector helpers have been ported over now?

**Review by TorreZuk — APPROVED** [review]
LGTM, good to see the full cpu reference.   Minor comments, but Daine should probably have a look as more familiar with the latest patterns.

File: clients/include/testing_syrkx_batched.hpp (hunk 2 above)
Comment (daineAMD) [reply clients/include/testing_syrkx_batched.hpp:89]: Yeah, the host_batched_vectors exist now. I don't mind going through with this PR now though, I've been making batched_vector changes along with other test changes to the entire test suite, I should be able to get to syrk in the next iteration.

**Review by daineAMD — APPROVED** [review]

**Pull request merged**